src/compiler/checker.ts

```diff
@@ -4499,7 +4525,19 @@ namespace ts {
                         if (source === numberType && target.flags & TypeFlags.Enum) return Ternary.True;
                     }
                 }
+
+                if (relation === assignableRelation && source.flags & TypeFlags.ObjectLiteral && source.flags & TypeFlags.FreshObjectLiteral) {

```

Comment:
Why check ObjectLiteral if all FreshObjectLiterals are ObjectLiterals? In fact, why not make the FreshObjectLiteral mask include the ObjectLiteral mask?